src/Core/Grid/Column/Type/HighlightedColumn.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Grid\Column\Type;
+
+use PrestaShop\PrestaShop\Core\Grid\Column\AbstractColumn;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * Displays data with highlighted column, the field option indicates which row
+ * field is displayed in this column The is_highlighted_field option indicates
+ * which row field enables/disables the highlighting
+ */
+final class HighlightedColumn extends AbstractColumn
```

Comment:
It looks very similar to BadgeColumn 🤔 ?

Reply:
That's true. There is only one tiny difference - it does not highlight(badge) button when cart is abandoned:

![image](https://user-images.githubusercontent.com/32385991/73728595-4afe2180-473c-11ea-9194-0dfd57262a01.png)

If that is not important when we can use BadgeColumn. So question is: Is it important to look the same as in LegacyController;

Reply:
You're right, highlighting abandoned carts would be bad from a UX point of view.

Alternatives are:
- dont highlight it (like you did)
- highlight it with another color (red ?) to acknowledge the difference

Can you please explore whether we can improve BadgeColumn in order to allow one of these 2 solutions to be implemented with BadgeColumn ? Basically I feel bad about having 2 columns _almost_ identical and only such a small difference. It looks like a duplicate mistake 😅 .

_However_ if you find out that, in order to use BadgeColumn for both its current usage and this one, you need to modify the code in such a way that it becomes very complex, then we'll stick with both BadgeColumn and HighlightColumn.

So, to sum it up: please try to modify BadgeColumn so it can be used in both usecases (= I guess we can do that by adding options), and if it's making the code too complex, then dont modify it and we'll keep both Columns.

Reply:
will do it, yeah, noticed that before changing anything, I kind of got used to discuss solution first

Reply:
Resolved with ifs inside badge.html.twig; Thought not the fan of the nested ifs solutions.